### PR TITLE
[#1590] docs: Add doc for column default value

### PR DESCRIPTION
**Pull request by mchades**
### What changes were proposed in this pull request?

Add doc for column default value

### Why are the changes needed?

Fix: #1590 

### Does this PR introduce _any_ user-facing change?

no

### How was this patch tested?

no need

### CODE REVIEW

File: docs/manage-metadata-using-gravitino.md [hunk 1]
```diff
@@ -790,6 +790,22 @@ The following types that Gravitino supports:
...
 The related java doc is [here](pathname:///docs/0.4.0/api/java/com/datastrato/gravitino/rel/types/Type.html).
 
+#### Table column default value
+You can specify a [literal](./expression.md#literal) or an [expression](./expression.md) as the default value when defining a table column. This default value is applied to new rows inserted into the table by the underlying catalog.
+
+When defining a table column, you can specify a [literal](./expression.md#field-reference) or an [expression](./expression.md) as the default value. The default value typically applies to new rows that are inserted into the table by the underlying catalog.
+
+:::note
+When Gravitino loads a table from a catalog that supports default value, if Gravitino is unable to parse the default value, it will use an [Unparsed Expression](./expression.md#unparsed-expression) to preserve the original default value, ensuring that the table can be loaded successfully.
+:::
```
Comment (jerryshao) [inline docs/manage-metadata-using-gravitino.md:800]: I think you should list some of the expressions you supported for column default value. Also point the "unparsed-expression" for others you don't support.

Comment (jerryshao) [reply docs/manage-metadata-using-gravitino.md:800]: Also update the case to add default value support here.

File: docs/expression.md [hunk 2]
```diff
@@ -133,3 +139,27 @@ FunctionExpression[] functionExpressions =
...
+  "type": "unparsed",
+  "unparsedExpression": "(curdate() + interval 1 year)"
+}
+```
+
+  </TabItem>
+  <TabItem value="java" label="Java">
+
+```java
+String unparsedValue = ((UnparsedExpression) expressino).unparsedExpression();
```
Comment (yuqi1129) [inline docs/expression.md:161]: Can you make the Java and JSON code consistent?

Comment (mchades) [reply docs/expression.md:161]: This code shows how to retrieve from a `UnparsedExpression`.  I will add a comment to the code.

File: docs/expression.md [hunk 3]
```diff
@@ -9,7 +9,7 @@ license: Copyright 2024 Datastrato Pvt Ltd. This software is licensed under the
 import Tabs from '@theme/Tabs';
 import TabItem from '@theme/TabItem';
 
-This page introduces the expression system of Gravitino. Expressions are vital component of metadata definition, through expressions, you can define default values for columns and function arguments for [function partitioning](./table-partitioning-bucketing-sort-order-indexes.md#table-partitioning), [bucketing](./table-partitioning-bucketing-sort-order-indexes.md#table-bucketing), and sort term of [sort ordering](./table-partitioning-bucketing-sort-order-indexes.md#sort-ordering) in tables.
+This page introduces the expression system of Gravitino. Expressions are vital component of metadata definition, through expressions, you can define [default values](./manage-metadata-using-gravitino.md#table-column-default-value) for columns and function arguments for [function partitioning](./table-partitioning-bucketing-sort-order-indexes.md#table-partitioning), [bucketing](./table-partitioning-bucketing-sort-order-indexes.md#table-bucketing), and sort term of [sort ordering](./table-partitioning-bucketing-sort-order-indexes.md#sort-ordering) in tables.
```
[the remark this answers is not in the document]
Comment (mchades) [reply docs/expression.md]: fixed

File: docs/expression.md [hunk 4]
```diff
@@ -133,3 +139,27 @@ FunctionExpression[] functionExpressions =
   </TabItem>
 </Tabs>
 
+## Unparsed expression
+
+An unparsed expression, a special type of expression, currently serves exclusively to represent the default value of a column when it's unsolvable.
```
[the remark this answers is not in the document]
Comment (mchades) [reply docs/expression.md]: fixed

File: docs/manage-metadata-using-gravitino.md (hunk 1 above)
Comment (mchades) [reply docs/manage-metadata-using-gravitino.md]: fixed

File: docs/jdbc-mysql-catalog.md [hunk 5]
```diff
@@ -107,59 +107,10 @@ MySQL doesn't support Gravitino `Boolean` `Fixed` `Struct` `List` `Map` `Timesta
...
-    Column.of("id", Types.IntegerType.get(), "id column comment", false, true, null),
-    Column.of("name", Types.VarCharType.of(500), "Name of the user", true, false, null)
-};
-Index[] indexes = new Index[] {
-    Indexes.of(IndexType.PRIMARY_KEY, "PRIMARY", new String[][]{{"id"}})
-}
-```
-
-</TabItem>
-</Tabs>
```
Comment (jerryshao) [inline docs/jdbc-mysql-catalog.md:161]: This is a special case for mysql auto increment, you should not delete it.

Comment (mchades) [reply docs/jdbc-mysql-catalog.md]: restored

File: docs/jdbc-mysql-catalog.md [hunk 6]
```diff
@@ -107,8 +107,6 @@ MySQL doesn't support Gravitino `Boolean` `Fixed` `Struct` `List` `Map` `Timesta
 
 #### Table column auto-increment
 
-- Supports setting auto-increment.
-
 :::notice
```
Comment (jerryshao) [inline docs/jdbc-mysql-catalog.md:110]: should be change to "note"?

Comment (mchades) [reply docs/jdbc-mysql-catalog.md]: yes, fixed

**Pull request merged**